pkg/upgrade/upgrade_suit_test.go

```diff
@@ -0,0 +1,175 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/stretchr/testify/mock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/Mellanox/network-operator/pkg/upgrade"
+	"github.com/Mellanox/network-operator/pkg/upgrade/mocks"
+	// +kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var k8sClient client.Client
+var k8sInterface kubernetes.Interface
+var testEnv *envtest.Environment
+var log logr.Logger
+var nodeUpgradeStateProvider mocks.NodeUpgradeStateProvider
+var drainManager mocks.DrainManager
+var podDeleteManager mocks.PodDeleteManager
+var uncordonManager mocks.UncordonManager
+
+var createdObjects []client.Object
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{}
+
+	cfg, err := testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	// +kubebuilder:scaffold:scheme
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	k8sInterface, err = kubernetes.NewForConfig(cfg)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sInterface).NotTo(BeNil())
+
+	log = ctrl.Log.WithName("upgradeSuitTest")
+
+	nodeUpgradeStateProvider = mocks.NodeUpgradeStateProvider{}
+	nodeUpgradeStateProvider.
+		On("ChangeNodeUpgradeState", mock.Anything, mock.Anything, mock.Anything).
+		Return(func(ctx context.Context, node *corev1.Node, newNodeState string) error {
+			node.Annotations[upgrade.UpgradeStateAnnotation] = newNodeState
+			return nil
+		})
+
+	drainManager = mocks.DrainManager{}
+	drainManager.
+		On("ScheduleNodesDrain", mock.Anything, mock.Anything).
+		Return(func(context.Context, *upgrade.DrainConfiguration) error {
```

Comment:
Return(nil) doesnt work here ?

the mocks support either returning specific values or using method as you did here.
if there is no special logic you want to do, then you can simplify and just use the desired return value explicitly.

I see you did that across all tests.

Reply:
Thanks, it works